modules/c++/sys/source/OSUnix.cpp

```diff
@@ -189,6 +189,7 @@ std::string sys::OSUnix::getTempName(const std::string& path,
     else
     {
         name = tempname;
+        fclose(fopen(name));
```

Comment:
The motivation is to make this consistent with Windows?  Could you just construct an unnamed `sys::File` here to get the same effect?

If creating the file is a side effect, update the AbstractOS documentation for this function.